src/iface/ethernet.rs

```diff
@@ -65,6 +65,8 @@ struct InterfaceInner<'b, 'c> {
     ip_addrs:               ManagedSlice<'c, IpCidr>,
     #[cfg(feature = "proto-ipv4")]
     ipv4_gateway:           Option<Ipv4Address>,
+    #[cfg(feature = "proto-ipv6")]
+    ipv6_gateway:           Option<Ipv6Address>,
```

Comment:
This works for now, but we really need to add support for NDISC Router Advertisements and Solicitations. In IPv6 this should actually be `default_routes` or something like that, where some could have been statically assigned and some could have been dynamically acquired via a Router Advertisement.

Reply:
Done, I added support for routes, to both IPv4 and IPv6.